xled/discover.py

```diff
@@ -432,6 +432,21 @@ def _next_packet(self):
             data, host = self.udp.recv(64)
             return data, host
 
+    def get_mac_address(self, ip_address):
+        ip = ip_address.decode("utf-8")
+
+        base_url = "http://{}/xled/v1/gestalt".format(ip)
+        r = requests.get(base_url)
+        if r.status_code != 200:
+            log.error("Failure getting MAC address from device at {}. Not a Twinkly?".format(ip))
+            return None
+
+        try:
+            hw_address = bytes(r.json().get("mac"), "utf-8")
```

Comment:
This breaks on python2:
```
Traceback (most recent call last):
  File ".../xled/xled/discover.py", line 445, in get_mac_address
TypeError: str() takes at most 1 argument (2 given)
```

either here or in `handle_beacon` differentiate between python versions.

Reply:
Well, with #57 in mind.